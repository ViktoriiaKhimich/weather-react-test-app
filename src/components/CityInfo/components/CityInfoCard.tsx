import React, { FC } from 'react';
import { Card, CardMedia, CardContent, Typography } from '@mui/material';
import ThermostatOutlinedIcon from '@mui/icons-material/ThermostatOutlined';

import { getCurrentDateAndTime } from '../../../helpers/dateAndTime/getCurrentDateAndTime';
import { convertUnixToTime } from '../../../helpers/dateAndTime/convertUnixToTime';
import { ICity } from '../../../interfaces';

interface IProps {
  city: ICity;
}

export const CityInfoCard: FC<IProps> = ({ city }) => {
  return (
    <Card
      variant='outlined'
      data-testid='cityInfoCard'
    >
      <CardContent>
        <Typography
          sx={{ fontSize: 14 }}
          color='text.secondary'
          gutterBottom
        >
          {getCurrentDateAndTime()}
        </Typography>
        <Typography
          variant='h3'
          component='div'
        >
          {city?.name}, {city?.sys.country}
        </Typography>
        <Typography
          style={{ display: 'flex' }}
          sx={{ mb: 1.5 }}
          color='text.secondary'
          component='div'
        >
          <CardMedia
            style={{ width: 50, height: 90, marginRight: 10 }}
            component='img'
            image={`http://openweathermap.org/img/w/${city?.weather[0].icon}.png`}
            alt='weather'
          />
          <div style={{ display: 'flex', alignItems: 'center' }}>
            <ThermostatOutlinedIcon />
            <h1>{city?.main.temp}째C</h1>
          </div>
        </Typography>
        <Typography
          sx={{ fontSize: 14 }}
          variant='body2'
          component='div'
        >
          <p style={{ display: 'flex', alignItems: 'center' }}>
            Feels like {city?.main.feels_like}째C. {city?.weather[0].main}
          </p>
          <p style={{ display: 'flex', alignItems: 'center' }}>
            Temperature is from {city?.main.temp_min}째C to {city?.main.temp_max}
            째C
          </p>
          <div
            style={{
              display: 'flex',
              justifyContent: 'space-between',
              marginTop: 20,
            }}
          >
            <div
              style={{
                display: 'flex',
                flexDirection: 'column',
                alignItems: 'start',
              }}
            >
              <p>Pressure: {city?.main.pressure}</p>
              <p>Humidity: {city?.main.humidity}</p>
              <p>Wind speed is: {city?.wind.speed} m/s</p>
            </div>
            <div
              style={{
                display: 'flex',
                flexDirection: 'column',
                alignItems: 'end',
                justifyContent: 'end',
              }}
            >
              <p>The sunrise is at {convertUnixToTime(city?.sys.sunrise)}</p>
              <p>The sunset is at {convertUnixToTime(city?.sys.sunset)}</p>
            </div>
          </div>
        </Typography>
      </CardContent>
    </Card>
  );
};
