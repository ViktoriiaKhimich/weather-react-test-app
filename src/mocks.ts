import { ICity, IDailyWeather } from './interfaces';
import { IWeatherState } from './store/weather/weatherSlice';
import { INotif } from './store/notificationSlice';

interface IAppState {
  weather: IWeatherState;
  notif: INotif;
}

export const citiesMock: ICity[] = [
  {
    coord: {
      lon: -0.1257,
      lat: 51.5085,
    },
    weather: [
      {
        id: 803,
        main: 'Clouds',
        description: 'broken clouds',
        icon: '04d',
      },
    ],
    base: 'stations',
    main: {
      temp: 19.22,
      feels_like: 19.05,
      temp_min: 18.25,
      temp_max: 20.56,
      pressure: 1031,
      humidity: 71,
    },
    visibility: 10000,
    wind: {
      speed: 3.6,
      deg: 330,
      gust: 9.24,
    },
    clouds: {
      all: 75,
    },
    dt: 1657353902,
    sys: {
      type: 2,
      id: 2019646,
      country: 'GB',
      sunrise: 1657338843,
      sunset: 1657397816,
    },
    timezone: 3600,
    id: 2643743,
    name: 'London',
    cod: 200,
  },
];

export const dailyWeatherMocks: IDailyWeather = {
  lat: 49.2497,
  lon: -123.1193,
  timezone: 'America/Vancouver',
  timezone_offset: -25200,
  daily: [
    {
      dt: 1657569600,
      sunrise: 1657541956,
      sunset: 1657599379,
      moonrise: 1657594620,
      moonset: 1657532400,
      moon_phase: 0.42,
      temp: {
        day: 21.29,
        min: 15.09,
        max: 23.17,
        night: 17.8,
        eve: 21.69,
        morn: 17.71,
      },
      pressure: 1020,
      humidity: 61,
      dew_point: 12.84,
      wind_speed: 3.15,
      wind_deg: 256,
      wind_gust: 2.78,
      feels_like: {
        day: 21.07,
        night: 17.59,
        eve: 21.43,
        morn: 17.47,
      },
      clouds: 0,
      pop: 0,
      uvi: 8,
      weather: [
        {
          id: 800,
          main: 'Clear',
          description: 'clear sky',
          icon: '01d',
        },
      ],
    },
  ],
};

export const mockAppState: IAppState = {
  weather: {
    cities: citiesMock,
    loading: false,
    error: null,
    dailyWeather: dailyWeatherMocks,
  },
  notif: {
    severity: '',
    message: '',
    isOpen: false,
  },
};

export const appState: IAppState = {
  weather: {
    cities: citiesMock,
    loading: true,
    error: null,
    dailyWeather: dailyWeatherMocks,
  },
  notif: {
    severity: 'error',
    message: 'City not found!',
    isOpen: true,
  },
};
