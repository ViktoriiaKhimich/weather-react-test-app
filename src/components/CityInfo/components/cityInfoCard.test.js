import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { CityInfoCard } from './CityInfoCard';
import { citiesMock } from '../../../mocks';

afterEach(cleanup);

it('renders correctly', () => {
  const { asFragment } = render(<CityInfoCard city={citiesMock[0]} />);
  expect(asFragment()).toMatchSnapshot();
});
