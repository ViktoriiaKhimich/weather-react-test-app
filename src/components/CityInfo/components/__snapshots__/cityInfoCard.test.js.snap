// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
    data-testid="cityInfoCard"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
      >
        July 13, 10:48 AM
      </p>
      <div
        class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
      >
        London, GB
      </div>
      <div
        class="MuiTypography-root MuiTypography-body1 css-9eg9l8-MuiTypography-root"
        style="display: flex;"
      >
        <img
          alt="weather"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
          src="http://openweathermap.org/img/w/04d.png"
          style="width: 50px; height: 90px; margin-right: 10px;"
        />
        <div
          style="display: flex; align-items: center;"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ThermostatOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15 13V5c0-1.66-1.34-3-3-3S9 3.34 9 5v8c-1.21.91-2 2.37-2 4 0 2.76 2.24 5 5 5s5-2.24 5-5c0-1.63-.79-3.09-2-4zm-4-8c0-.55.45-1 1-1s1 .45 1 1h-1v1h1v2h-1v1h1v2h-2V5z"
            />
          </svg>
          <h1>
            19.22째C
          </h1>
        </div>
      </div>
      <div
        class="MuiTypography-root MuiTypography-body2 css-5hrm3t-MuiTypography-root"
      >
        <p
          style="display: flex; align-items: center;"
        >
          Feels like 19.05째C. Clouds
        </p>
        <p
          style="display: flex; align-items: center;"
        >
          Temperature is from 18.25째C to 20.56째C
        </p>
        <div
          style="display: flex; justify-content: space-between; margin-top: 20px;"
        >
          <div
            style="display: flex; flex-direction: column; align-items: start;"
          >
            <p>
              Pressure: 1031
            </p>
            <p>
              Humidity: 71
            </p>
            <p>
              Wind speed is: 3.6 m/s
            </p>
          </div>
          <div
            style="display: flex; flex-direction: column; align-items: end; justify-content: end;"
          >
            <p>
              The sunrise is at 6:54 AM
            </p>
            <p>
              The sunset is at 11:16 PM
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
